### PR TITLE
Backup & Restore functions for the firestore

### PR DESCRIPTION
**Pull request by lsenta**

### CODE REVIEW

File: functions/src/backup.ts [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+type QueryDocumentSnapshot = admin.firestore.QueryDocumentSnapshot;
+type DocumentReference = admin.firestore.DocumentReference;
+
+interface StoredDocument {
+  docPath: string;
+  content: any;
+}
+
+class Queue {
+  content: string[];
```
Comment (GrandSchtroumpf) [inline functions/src/backup.ts]: set the initial value here : `content: string[] = []`

File: functions/src/users.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  id: string,
+  email: string,
+}
+
+interface OrgProposal {
+  id: string,
+  name: string,
+}
+
+const onUserCreate = (user: UserRecord) => {
```
Comment (GrandSchtroumpf) [inline functions/src/users.ts]: We do this client side right now. Any reason why to do it inside a function ?

File: functions/src/users.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  const incLast: string = String.fromCharCode(prefix.slice(-1).charCodeAt(0) + 1);
+  const prefixEnd: string = prefix.slice(0, -1) + incLast;
+
+  return db.collection('users')
+    .where('email', '>=', prefix)
+    .where('email', '<', prefixEnd)
+    .get()
+    .then(q => {
+      // leave if there are too many results.
+      if (q.size > 10) {
```
Comment (GrandSchtroumpf) [inline functions/src/users.ts]: Why ?

File: functions/src/users.ts (hunk 2 above)
Comment (lsenta) [reply functions/src/users.ts]: More reliable to do this server side, that's legacy btw, I just structure the code to clean the index.ts

File: functions/src/users.ts (hunk 3 above)
Comment (lsenta) [reply functions/src/users.ts]: Avoid showing the whole database to a user

Comment (GrandSchtroumpf) [reply functions/src/users.ts]: Then why returning an empty array instead of the 10 first ?

**Pull request merged**